@utahdts/utah-design-system/react/components/buttons/ConfirmationButton.jsx

```diff
@@ -58,29 +87,95 @@ function ConfirmationButton({
   promptChildren,
   size,
   type,
+  ...rest
 }) {
   const [isClicked, setIsClicked] = useState(false);
-  const buttonClick = () => {
-    setIsClicked(!isClicked);
-    onClick();
-  };
+  const timer = useRef(NaN);
+  const { addAssertiveMessage } = useAriaMessaging();
+
+  const clearTimer = useCallback(() => {
+    if (timer.current) {
+      clearTimeout(timer.current);
+      timer.current = NaN;
+    }
+  }, []);
+
+  const resetButton = useCallback(() => {
+    clearTimer();
+    setIsClicked(false);
+    addAssertiveMessage(children);
+  }, [addAssertiveMessage, children, clearTimer]);
+
+  const handleOnClick = useCallback((e) => {
+    if (!isBusy) {
+      if (isClicked) {
+        onClick?.(e);
+        resetButton();
+      } else {
+        clearTimer();
+        setIsClicked(true);
+        addAssertiveMessage(promptChildren);
+        timer.current = setTimeout(resetButton, 3500);
+      }
+    }
+  }, [addAssertiveMessage, clearTimer, isBusy, isClicked, onClick, promptChildren, resetButton]);
+
+  const renderLeftIcon = useCallback(() => {
+    let icon = null;
+    if (iconLeft?.initial && !isClicked) {
+      icon = <span className="button--icon button--icon-left">{iconLeft.initial}</span>;
+    } if (iconLeft?.prompt && isClicked) {
+      icon = <span className="button--icon button--icon-left">{iconLeft.prompt}</span>;
+    }
+    return icon;
+  }, [iconLeft, isClicked]);
+
+  const renderRightIcon = useCallback(() => {
+    let icon = null;
+    if (iconRight?.initial && !isClicked) {
+      icon = <span className="button--icon button--icon-right">{iconRight.initial}</span>;
+    } if (iconRight?.prompt && isClicked) {
+      icon = <span className="button--icon button--icon-right">{iconRight.prompt}</span>;
+    }
+    return icon;
+  }, [iconRight, isClicked]);
+
   return (
-    <Button
-      appearance={appearance}
-      className={className}
-      color={color}
-      innerRef={innerRef}
-      isBusy={isBusy}
-      iconLeft={iconLeft}
-      iconRight={iconRight}
-      isDisabled={isDisabled}
+    <button
+      className={joinClassNames(
+        'button',
+        className,
+        `button--${appearance}`,
+        // default color is none
+        (color && color !== 'none') ? `button--${color}-color` : null,
+        // default size is medium
+        (size && size !== formElementSizesEnum.MEDIUM) ? `button--${size}` : null,
+        isClicked ? 'button--confirm' : null
+      )}
+      disabled={isDisabled || isBusy}
       id={id}
-      onClick={buttonClick}
-      size={size}
+      ref={innerRef}
+      onClick={handleEvent(handleOnClick)}
```

Comment:
does handleevent use callback? might be better to put the handleeveent in the usecallback?

Reply:
Refer to comment above code example using HandleEvent.